### PR TITLE
Asset improvements

### PR DESCRIPTION
**Pull request by MitjaBezensek**
This PR does the following:
- Add `selfHosted.js`, which is a great option for users that wish to self host the assets. Works well for both self hosting from the public folder or via a CDN.
- Updates the docs for assets. We now have a dedicated page for assets where all the options are more clearly explained. I also removed the assets explanation from the main docs as the unpkg option should work out of the box and setting up the assets is no longer necessary.
- Cleaned up the `refresh-assets` script. We now use common `types.d.ts` file to define our types. All the other options then reuse them.
- Pulled out the `formatAssetUrl` into it's own file. It's now static an no longer generated.
- `urls.d.ts`, `import.d.ts`, and newly added `selfhosted.d.ts` are now also no longer generated as we can import the types from `types.d.ts`.
- You can now pass a subset of `assetUrls` to `<Tldraw />` and it will override the default option with the passed in overrides. This makes it easy to only customizes certain assets (only change the draw font as an example).

### Change Type

- [x] `patch` — Bug Fix

### CODE REVIEW

File: packages/assets/imports.d.ts [hunk 1]
```diff
@@ -3,8 +3,7 @@
 
 type AssetUrl = string | { src: string }
 type AssetUrlOptions = { baseUrl?: string } | ((assetUrl: string) => string)
-
-export function getAssetUrlsByImport(opts?: AssetUrlOptions): {
+type AssetUrls = {
```
Comment (MitjaBezensek) [inline packages/assets/imports.d.ts:6]: Could even reuse this between files.

File: packages/assets/urls.js [hunk 2]
```diff
@@ -644,3 +644,235 @@ export function getAssetUrlsByMetaUrl(opts) {
 		},
 	}
 }
+
+export function getAssetUrls(opts) {
```
Comment (SomeHats) [inline packages/assets/urls.js:648]: what if we write this into a `local.js`? I think this file might cause errors if you import it from a bundler that doesn't support the special meta url syntax. So you would do `import {getAssetUrls} from '@tldraw/assets/local'` or something??? maybe local isn't the best name though i'm not sure

**Comment by MitjaBezensek** [comment]
I have read and agree to the Contributor License Agreement.

File: packages/assets/imports.d.ts [hunk 3]
```diff
@@ -1,232 +1,2 @@
...
-		observable: string
-		replit: string
-		scratch: string
-		spotify: string
-		tldraw: string
-		vimeo: string
-		youtube: string
-	}
-}
+import { AssetUrlOptions, AssetUrls } from './types'
```
Comment (MitjaBezensek) [inline packages/assets/imports.d.ts:1]: Types are now reused, so these `.d.ts` are no longer generated. We only generate `types.d.ts`.

File: packages/assets/urls.js (hunk 2 above)
Comment (MitjaBezensek) [reply packages/assets/urls.js]: Used `selfHosted` name, but also unsure about it. Decided against local as this option is also quite useful for hosting via a CDN (you just pass in CDN's base url), so local didn't feel right for that.

File: apps/docs/content/docs/assets.mdx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+const assetUrls = getAssetUrls({
+    baseUrl: 'https://unpkg.com/@tldraw/assets@2.0.0-alpha.12/',
+})
+```
+
+### 3. Using a bundler
+
+If you're using a bundler like webpack or rollup, you can import the assets directly from the `@tldraw/assets` package. Here you can use `getAssetUrlsByMetaUrl` helper function:
+```javascript
+import { getAssetUrlsByMetaUrl } from '@tldraw/assets/selfHosted'
```
Comment (SomeHats) [inline apps/docs/content/docs/assets.mdx:60]: should be `@tldraw/assets/urls`

File: apps/docs/content/docs/assets.mdx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+```javascript
+module: {
+    rules: [
+        {
+            test: /\.(svg|json|woff2)$/,
+            type: 'asset/resource'
+        }
+    ]
+}
+ ```
```
Comment (SomeHats) [inline apps/docs/content/docs/assets.mdx:77]: this isn't actually needed with the `urls` import - it's only for the `import` one

**Review by SomeHats — APPROVED** [review]
love these docs changes! thanks mitja :D

File: apps/docs/content/docs/assets.mdx (hunk 4 above)
Comment (MitjaBezensek) [reply apps/docs/content/docs/assets.mdx]: That's what I get for copy pasting 😂

**Pull request merged**